homeassistant/components/reolink/entity.py

```diff
@@ -155,6 +155,11 @@ def __init__(
                 configuration_url=self._conf_url,
             )
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._host.api.camera_online(self._channel) and super().available
```

Comment:
Shouldn't this be the other way around?

```suggestion
        return super().available and self._host.api.camera_online(self._channel)
```

Reply:
Does that matter?
It is a 'and' condition so that is reverseble right?

But sure, the other way around is also good.

Reply:
> t is a 'and' condition so that is reverseble right?

No, it is not.

It will stop on the first false. So, in case the coordinator is unavailable, it will not run/evaluate the second statement.

Reply:
Oh right, if self._host.api.camera_online would have been a API call, you could safe yourself the time of the API call in case the first is already false....

But in this case `self._host.api.camera_online` is cached so it will return emidiatly, the values are updates when calling `self._host.api.get_states()` which is done in the UpdateCoordinator.

Reply:
Good too keep in mind, thanks Frenck